pkg/kubectl/cmd/get/get.go

```diff
@@ -224,9 +225,12 @@ func (o *GetOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []stri
 
 	o.IncludeUninitialized = cmdutil.ShouldIncludeUninitialized(cmd, false)
 
-	if resource.MultipleTypesRequested(args) {
+	discoveryClient, discoveryErr := f.ToDiscoveryClient()
+	if resource.MultipleTypesRequested(args) ||
+		(discoveryErr == nil && resource.AnyCategoryRequested(restmapper.NewDiscoveryCategoryExpander(discoveryClient), args)) {
```

Comment:
The logic in `MultipleTypesRequested` still contains special casing for the `all` category. We can remove the special casing if we move the categories check inside that function.

Reply:
Or we can remove the special casing there, but I guess it's better to have it all in the function?

Reply:
it would make more sense to remove the check from Complete(), and figure out how to get the info about whether multiple types were requested from the component that actually knows already, which is the resource builder in `Run()`, after `ResourceTypeOrNameArgs(...)` has been called